NDTensors/src/default_storage.jl

```diff
@@ -0,0 +1,19 @@
+## This is a fil which specifies the default storage type provided some set of parameters
+## The parameters are the element type and storage type
+default_datatype(eltype::Type{<:Number}) = Vector{eltype}
+default_eltype() = Float64
+
+default_storagetype(datatype::Type{<:AbstractArray{ElT}}) where {ElT} = Dense{ElT, datatype}
```

Comment:
I prefer to write this as:
```julia
default_storagetype(datatype::Type{<:AbstractArray}) = Dense{eltype(datatype), datatype}
```
In general I think it is best to use functions like that instead of `where` as much as possible, both as a style choice and also it can make it easier to write generic code.

Reply:
I'll try to reconfigure and move forward without using where (when possible)